src/classes/RD2_EntryFormController.cls

```diff
@@ -136,13 +136,15 @@ public with sharing class RD2_EntryFormController {
         List<Schema.FieldSetMember> fieldSetFields = Schema.SObjectType.npe03__Recurring_Donation__c.fieldSets.RD2_Entry_Form_Custom_Fields.getFields();
 
         for (Schema.FieldSetMember field : fieldSetFields) {
-            if (field.getFieldPath().startsWith('npsp__')
+            if (!UTIL_Describe.getFieldDescribe('npe03__Recurring_Donation__c', field.getFieldPath()).isAccessible()
```

Comment:
Just to help slightly with performance I recommend re-sequencing these slightly:
1. Excluded fields list
2. StartsWith
3. IsAccessible

This way it's only doing a field describe on fields that meet the other two conditions first.